internal/node/controller/auth.go

```diff
@@ -103,3 +109,186 @@ func getUserCertAndInfo(controller NodeController, username string) (*x509.Certi
 
 	return cert, user, nil
 }
+
+// BFF Auth - Allow for Habitat node's to authenticate with each other.
+
+type BFFChallengeRequest struct {
+	DID string `json:"did"`
+}
+
+type FriendStore map[string]*Friend
+
+type Friend struct {
+	DID       string `json:"did"`
+	PublicKey crypto.PublicKey
+}
+
+type FriendRequest struct {
+	DID                string `json:"did"`
+	PublicKeyMultibase string `json:"public_key_multibase"`
+}
+
+type BFFProvider struct {
+	challengePersister bffauth.ChallengeSessionPersister
+	friends            FriendStore
+	signingKey         []byte
+}
+
+func NewBFFProvider(challengePersister bffauth.ChallengeSessionPersister, friends FriendStore, signingKey []byte) *BFFProvider {
+	return &BFFProvider{
+		challengePersister: challengePersister,
+		friends:            friends,
+		signingKey:         signingKey,
+	}
+}
+
+func (p *BFFProvider) GetRoutes() []api.Route {
+	return []api.Route{
+		newRoute(http.MethodPost, "/node/bff/challenge", p.handleChallenge),
+		newRoute(http.MethodPost, "/node/bff/auth", p.handleAuth),
+		newRoute(http.MethodPost, "/node/bff/add_friend", p.handleAddFriend),
+		newRoute(http.MethodGet, "/node/bff/test", p.handleTest),
+	}
+}
+
+func (p *BFFProvider) handleChallenge(w http.ResponseWriter, r *http.Request) {
+	var req BFFChallengeRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	_, ok := p.friends[req.DID]
+	if !ok {
+		http.Error(w, fmt.Sprintf("Friend with DID %s not found", req.DID), http.StatusNotFound)
+		return
+	}
+
+	challenge, err := bffauth.GenerateChallenge()
+	if err != nil {
+		http.Error(w, "Failed to generate challenge", http.StatusInternalServerError)
+		return
+	}
+
+	// Create session with UUID
+	sessionID := uuid.New().String()
+	session := &bffauth.ChallengeSession{
+		SessionID: sessionID,
+		DID:       req.DID,
+		Challenge: challenge,
+		ExpiresAt: time.Now().Add(15 * time.Minute),
+	}
+	err = p.challengePersister.SaveSession(session)
+	if err != nil {
+		http.Error(w, "Failed to save session", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(map[string]string{
```

Comment:
same thing here I think `Encoder` is meant for strings -- can we just do `json.Marshal`

Reply:
It's fine if you don't do in this PR, I'm going to take a pass and standardize all of these. Same with our testing.